plugins/destination/plugin_unimplemented.go

```diff
@@ -0,0 +1,65 @@
+package destination
+
+import (
+	"context"
+	"time"
+
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/cloudquery/plugin-pb-go/specs"
+	"github.com/cloudquery/plugin-sdk/v3/schema"
+)
+
+type UnimplementedUnmanagedWriter struct{}
+
+func (UnimplementedUnmanagedWriter) Write(context.Context, specs.Source, schema.Tables, time.Time, <-chan arrow.Record) error {
+	panic("Write not implemented")
+}
+
+func (UnimplementedUnmanagedWriter) Metrics() Metrics {
+	panic("Metrics not implemented")
+}
+
+type UnimplementedManagedWriter struct{}
+
+func (UnimplementedManagedWriter) WriteTableBatch(context.Context, specs.Source, *schema.Table, time.Time, []arrow.Record) error {
+	panic("WriteTableBatch not implemented")
+}
+
+type UnimplementedRead struct{}
+
+func (UnimplementedRead) Read(context.Context, *schema.Table, string, chan<- arrow.Record) error {
+	panic("Read not implemented")
+}
+
+type UnimplementedMigrate struct{}
+
+func (UnimplementedMigrate) Migrate(context.Context, schema.Tables) error {
+	return nil // Special case, we don't want to error here
+}
+
+type UnimplementedDeleteStale struct{}
+
+func (UnimplementedDeleteStale) DeleteStale(context.Context, schema.Tables, string, time.Time) error {
+	return nil // Special case, we don't want to error here
+}
+
+type UnimplementedClose struct{}
+
+func (UnimplementedClose) Close(context.Context) error {
+	return nil // Special case, we don't want to error here
+}
+
+type UnimplementedClient struct {
```

Comment:
We can just use a single `UnimplementedClient` empty struct and move all functions to it.
This way we simply embed it into the destination client and implement only the required methods.

Reply:
I don't remember why I separated them, but some of the functionality shouldn't error when not implemented, maybe that was the reason behind this... I'll refactor once we get some more reviews of that's OK. (We could also do an ErrNotImplemented type thing and then make the SDK ignore those errors maybe)